applications/FluidDynamicsApplication/custom_strategies/strategies/compressible_navier_stokes_explicit_solving_startegy.h

```diff
@@ -0,0 +1,576 @@
+     * @param rModelPart The model part of the problem
+     * @param ThisParameters The configuration parameters
+     */
+    explicit CompressibleNavierStokesExplicitSolvingStrategy(
+        ModelPart &rModelPart,
+        Parameters ThisParameters)
+        : BaseType(rModelPart)
+    {
+        KRATOS_TRY
+
+        // Validate and assign defaults
+        ThisParameters = this->ValidateAndAssignParameters(ThisParameters, this->GetDefaultParameters());
+        this->AssignSettings(ThisParameters);
+
+        KRATOS_CATCH("")
+    }
+
+    /**
+     * @brief Default constructor.
+     * @param rModelPart The model part to be computed
+     * @param pExplicitBuilder The pointer to the explicit builder and solver
+     * @param MoveMeshFlag The flag to set if the mesh is moved or not
+     */
+    explicit CompressibleNavierStokesExplicitSolvingStrategy(
+        ModelPart &rModelPart,
+        typename ExplicitBuilderType::Pointer pExplicitBuilder,
+        bool MoveMeshFlag = false,
+        int RebuildLevel = 0)
+        : BaseType(rModelPart, pExplicitBuilder, MoveMeshFlag, RebuildLevel)
+    {
+    }
+
+    /**
+     * @brief Default constructor.
+     * @param rModelPart The model part to be computed
+     * @param MoveMeshFlag The flag to set if the mesh is moved or not
+     */
+    explicit CompressibleNavierStokesExplicitSolvingStrategy(
+        ModelPart &rModelPart,
+        bool MoveMeshFlag = false,
+        int RebuildLevel = 0)
+        : BaseType(rModelPart, MoveMeshFlag, RebuildLevel)
+    {
+    }
+
+    /** Copy constructor.
+     */
+    CompressibleNavierStokesExplicitSolvingStrategy(const CompressibleNavierStokesExplicitSolvingStrategy &Other) = delete;
+
+    /** Destructor.
+     */
+    virtual ~CompressibleNavierStokesExplicitSolvingStrategy() = default;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    int Check() override
+    {
+        int err_code = BaseType::Check();
+
+        // Check that the process info already contains the OSS activation variable
+        const auto& r_process_info = BaseType::GetModelPart().GetProcessInfo();
+        KRATOS_ERROR_IF_NOT(r_process_info.Has(OSS_SWITCH)) << "OSS_SWITCH variable has not been set in the ProcessInfo container. Please set it in the strategy \'Initialize\'." << std::endl;
+
+        // Shock capturing process check
+        if (ShockCapturingEnabled()) {
+            GetShockCapturingProcess()->Check();
+        }
+
+        return err_code;
+    }
+
+    /**
+     * @brief This method provides the defaults parameters to avoid conflicts between the different constructors
+     * @return The default parameters
+     */
+    Parameters GetDefaultParameters() const override
+    {
+        KRATOS_TRY
+
+        Parameters default_parameters = Parameters(R"(
+        {
+            "name" : "compressible_navier_stokes_explicit_explicit_solving_strategy_runge_kutta_4",
```

Comment:
This is not current class name. In other words, this field must be implemented by each class by calling the add missing parameters. Otherwise the factory mechanism we're working on will not work,

Reply:
Okay. What do we do in case of templated classes? For instance, RungeKutta. For now I used the Name() method.

Reply:
That's what I was wondering. The point is that either the `name` field or the `Name` method would be required to be defined in each one of the derived classes. Not sure how we can solve it...

Reply:
I'd say we derive Name in each of the classes, and templated classes should procedurally generate their own names depending on template parameters when relevant.

For instance, Runge Kutta generates its name by concatenating:
```
static std::string Name()
{
    return "compressible_navier_stokes_explicit_solving_strategy_runge_kutta_" + TButcherTableau::Name();
}
```

Maybe we can make CompressibleNavierStokesExplicitStrategy<...>::Name() virtual to force it to be derived?

Reply:
Makes sense for me. I think this would do the job.

Reply:
Well I just noticed Name() is static, so we cannot make it virtual, thus polymorphism won't help us here.

I think I'll go back to the good old recursive call to GetDefaultParameters()